source/edk.c

```diff
@@ -72,13 +72,10 @@ bool aws_cryptosdk_edk_is_valid(const struct aws_cryptosdk_edk *const edk) {
 }
 
 bool aws_cryptosdk_edk_list_elements_are_valid(const struct aws_array_list *edk_list) {
-    size_t len = edk_list->length;
-    for (size_t i = 0; i < len; ++i) {
-        struct aws_cryptosdk_edk *edk;
-        if (!aws_array_list_get_at_ptr(edk_list, (void **)&edk, i)) {
-            if (!aws_cryptosdk_edk_is_valid(edk)) {
-                return false;
-            }
+    for (size_t i = 0; i < edk_list->length; ++i) {
+        struct aws_cryptosdk_edk *data = (struct aws_cryptosdk_edk *)edk_list->data;
+        if (!aws_cryptosdk_edk_is_valid(&(data[i]))) {
+            return false;
```

Comment:
Is this change just a verification performance optimization?

Reply:
I was running into a CBMC error with the old implementation, so I changed it to this to match what we do in make_common_data_structures.c. I believe the actual CBMC error was caused elsewhere, but left the change in to be consistent with how we handle edk_lists elsewhere.